### PR TITLE
Feature/update events

### PR DESCRIPTION
**Pull request by codiebeulaine**
@miltontony ready for review

### CODE REVIEW

File: mentorship/mentor/models.py [hunk 1]
```diff
@@ -298,6 +301,28 @@ class CallNote(models.Model):
     created_at = models.DateTimeField(auto_now_add=True)
 
 
+@receiver(post_save, sender=CallNote,
+          dispatch_uid="create_event")
+def create_event(sender, instance, created, **kwargs):
```
Comment (miltontony) [inline mentorship/mentor/models.py]: This needs a better name - perhaps something like `create_event_on_call_note_post_save()`

File: mentorship/mentor/models.py [hunk 2]
```diff
@@ -298,6 +301,28 @@ class CallNote(models.Model):
...
+    if created:
+        Event.objects.create(
+            event_type='call_notes_created', description='Call notes created',
+            relevant_mentor=instance.mentor, object_id=instance.pk)
+    if instance.call.activity:
+        category = Category.objects.filter(
+            pk=instance.call.activity.parent).first()
```
Comment (miltontony) [inline mentorship/mentor/models.py]: this is simply `category = instance.call.activity.get_parent().specific`

File: mentorship/mentor/models.py [hunk 3]
```diff
@@ -298,6 +301,28 @@ class CallNote(models.Model):
...
+        activities = category.get_children().live()
+        all_activities_completed = True
+        for activity in activities:
+            if not activity.specific.is_complete(instance.mentor.user):
+                all_activities_completed = False
+        if all_activities_completed:
```
Comment (miltontony) [inline mentorship/mentor/models.py]: You can wrap the above for loop into the following
```python
if not any([for activity in activities if not activity.specific.is_complete(instance.mentor.user)]):
```
What this does is that it creates a list of articles that are **not complete** and then checks if the list is empty using `not any()`.

And example would be:
```
>>> fruits = [('apples', 4), ('pears', 0), ('mangos', 3)]
>>> fruits_finished = True
>>> for fruit, num in fruits:
...     if num > 0:
...         fruits_finished = False
...
>>>
>>> fruits_finished_new = not any([fruit for fruit, num in fruits if num > 0])
>>> print fruits_finished, fruits_finished_new
False False
>>>
>>> fruits = [('apples', 0), ('pears', 0), ('mangos', 0)]
>>> fruits_finished = True
>>> for fruit, num in fruits:
...     if num > 0:
...         fruits_finished = False
...
>>>
>>> fruits_finished_new = not any([fruit for fruit, num in fruits if num > 0])
>>> print fruits_finished, fruits_finished_new
True True
>>>
```

**Comment by miltontony** [comment]
👍 🍸

File: mentorship/mentor/models.py (hunk 3 above)
Comment (codiebeulaine) [inline mentorship/mentor/models.py]: @miltontony if not any([activity for activity in activities if not activity.specific.is_complete(instance.mentor.user)]):

made a small change

Comment (miltontony) [inline mentorship/mentor/models.py]: 👍

**Comment by justinvdm** [comment]
Post-merge :+1: 🍰

**Pull request merged**